### PR TITLE
Add a user guide in the README

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### Problems to solve

- Users were unaware of the aggregated proof example.
- Users were unaware of other repositories and didn't know which one to use in their project.

We further explain the nuances and features of the project in this README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+## User Guide
+
+This project is tailored for verifying [Halo2](https://github.com/privacy-scaling-explorations/halo2)-generated proofs on Ethereum.
+
+Additionally, within this repository, there is an example provided for verifying an aggregated proof, which serves as proof for multiple other proofs. You can test it by executing `cargo run --example evm-verifier-with-accumulator`.
+
+As this effort is continuously evolving, there are multiple repository variants to choose from. For optimal results:
+
+- For production use, it is recommended to use [axiom’s fork](https://github.com/axiom-crypto/snark-verifier), which has been audited.
+- In the future, consider using [halo2-solidity-verifier](https://github.com/privacy-scaling-explorations/halo2-solidity-verifier). However, please note the following:
```
Comment (han0110) [inline README.md:14]: This section seems only applicable for solidity verifier generation, we still need to use `snark-verifier` for proof aggregation even `halo2-solidity-verifier` is mature.

```suggestion
- In the future, consider using [halo2-solidity-verifier](https://github.com/privacy-scaling-explorations/halo2-solidity-verifier) for solidity verifier generation. However, please note the following:
```

**Review by han0110 — COMMENTED** [review]
LGTM! Thanks for adding this! Just a nitpick.

**Comment by ChihChengLiang** [comment]
@han0110 I've addressed the feedback and added the proof aggregation part.

**Review by han0110 — APPROVED** [review]
LGTM! Thanks for upgrading the doc!

**Pull request merged**